src/main/java/com/mraof/minestuck/entity/underling/ImpEntity.java

```diff
@@ -28,6 +29,9 @@
 
 public class ImpEntity extends UnderlingEntity implements IAnimatable
 {
+	public static final MobAnimationPhases CLAW_PHASES = new MobAnimationPhases(2, 4, 5, 10);
+	public static final MobAnimation CLAW_ANIMATION = new MobAnimation(MobAnimation.Actions.CLAW, CLAW_PHASES.getTotalAnimationLength(), true,false);
```

Comment:
looking at a typical usage, id put the animation phase inside the mob animation since it dosnt look used in the entity itself

Reply:
i meant that we could hide MobAnimationPhases completely in a class that inherits mobAnimation and let that create it if needed

Reply:
Seems reasonable!

Reply:
MobAnimationPhases is now PhasedMobAnimation and it extends MobAnimation, feels a lot cleaner now!